### PR TITLE
okurezgi/issue27

### LINKED ISSUE #27: PATCH /post/:id
This endpoint should update the requested post. 
The user should be able to edit their own posts only.

Add documentation and testing.

### PR DESCRIPTION
**Pull request by okurezgi**
Added controller and documentation for updating product by id

- lodash package added to the 'package.json'
- controller code added to the 'src/controllers/product.js'
- documentation with swagger added to the 'src/docs/product.yaml'
- 'publisher' added to the product schema 'src/models/product.js'
- routes added to the 'src/routes/product.js'

### CODE REVIEW

**Review by khatibAmjad — APPROVED** [review]
Kankim this code is sooo impressive

File: src/controllers/product.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    if (product.isTransactionCompleted) {
+      return res.status(400).json({
+        message:
+          'Transaction is already completed, you cannot update this product',
+      });
+    }
+
+    const validKeys = [
```
Comment (halitbatur) [inline src/controllers/product.js]: can we move this to a constant variable and import it from there?

File: src/controllers/product.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      'description',
+      'photos',
+      'category',
+      'location',
+      'productCondition',
+      'shippingOptions',
+      'postType',
+    ];
+
+    const updatedProduct = _.pick(req.body, validKeys);
```
Comment (awiednoor) [inline src/controllers/product.js]: I like how you used lodash here 💯

**Review by halitbatur — APPROVED** [review]

File: src/controllers/product.js (hunk 1 above)
Comment (okurezgi) [reply src/controllers/product.js]: sure thinggg

**Pull request merged**